lib/tasks/helpers/newrelicyml.rb

```diff
@@ -0,0 +1,142 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../new_relic/agent/configuration/default_source'
+
+module NewRelicYML
+  CRITICAL = [:'agent_enabled', :'app_name', :'license_key', :'log_level']
+  DEFAULTS = NewRelic::Agent::Configuration::DEFAULTS
+  # Skip because not configurable via yml
+  SKIP = [:'defer_rails_initialization']
+  # Don't evaluate Procs, instead use set values
+  PROCS = {:'config_path' => 'newrelic.yml',
+           :'process_host.display_name' => 'default hostname',
+           :'transaction_tracer.transaction_threshold' => 1.0}
+
+  HEADER = <<~HEADER
+    #
+    # This file configures the New Relic Agent.  New Relic monitors Ruby, Java,
+    # .NET, PHP, Python, Node, and Go applications with deep visibility and low
+    # overhead.  For more information, visit www.newrelic.com.
+
+    # Generated <%= Time.now.strftime('%B %d, %Y') %><%= ", for version \#{@agent_version}" if @agent_version %>
+    #<%= "\\n# \#{generated_for_user}\\n#" if generated_for_user %>
+    # For full documentation of agent configuration options, please refer to
+    # https://docs.newrelic.com/docs/agents/ruby-agent/installation-configuration/ruby-agent-configuration
+
+    common: &default_settings
+      # Required license key associated with your New Relic account.
+      license_key: <%= license_key %>
+
+      # Your application name. Renaming here affects where data displays in New
+      # Relic. For more details, see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/renaming-applications
+      app_name: <%= app_name %>
+
+      # To disable the agent regardless of other settings, uncomment the following:
+      # agent_enabled: false
+
+      # Logging level for log/newrelic_agent.log; options are error, warn, info, or
+      # debug.
+      log_level: info
+
+      # All of the following configuration options are optional. Review them, and
+      # uncomment or edit them if they appear relevant to your application needs.
+
+  HEADER
+
+  FOOTER = <<~FOOTER
+    # Environment-specific settings are in this section.
+    # RAILS_ENV or RACK_ENV (as appropriate) is used to determine the environment.
+    # If your application has other named environments, configure them here.
+    development:
+      <<: *default_settings
+      app_name: <%= app_name %> (Development)
+
+    test:
+      <<: *default_settings
+      # It doesn't make sense to report to New Relic from automated test runs.
+      monitor_mode: false
+
+    staging:
+      <<: *default_settings
+      app_name: <%= app_name %> (Staging)
+
+    production:
+      <<: *default_settings
+  FOOTER
+
+  def self.get_configs(defaults)
+    final_configs = {}
+
+    defaults.sort.each do |key, value|
+      next if CRITICAL.include?(key) || SKIP.include?(key)
+
+      if public_config?(value)
```

Comment:
It looks like there isn't an `else` scenario. Maybe this should be guarded with a `next unless` similar to the `next if` above it.

Reply:
I like - done! https://github.com/newrelic/newrelic-ruby-agent/pull/2075/commits/2d0e7236d726b06c6f8187d3aa08d92c083d5aed